packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -474,30 +548,108 @@ class TimeInput extends React.Component {
+  handleSecondInputKeyDown(event) {
+    let stateValue = this.state.second;
+    const previousStateValue = stateValue;
+
+    if (event.keyCode === KeyCode.KEY_UP) {
+      stateValue = TimeUtil.incrementSecond(stateValue);
+    }
+
+    if (event.keyCode === KeyCode.KEY_DOWN) {
+      stateValue = TimeUtil.decrementSecond(stateValue);
+    }
+
+    if (previousStateValue !== stateValue) {
+      this.handleValueChange(event, TimeUtil.inputType.SECOND, stateValue, this.state.meridiem);
+    }
+
+    if (event.keyCode === KeyCode.KEY_LEFT
+        || event.keyCode === KeyCode.KEY_DELETE
+        || event.keyCode === KeyCode.KEY_BACK_SPACE) {
+      this.focusMinuteFromSecond(event);
+    }
+
+    if (event.keyCode === KeyCode.KEY_RIGHT) {
+      this.focusMeridiemFromSecond(event);
+    }
+  }
+
+  focusMinuteFromSecond(event) {
+    // If the cursor is at the left most position in the second input, is empty or the cursor is before the value,
+    // move focus to the minute input
+
+    if (this.secondInput.selectionEnd === 0) {
+      this.minuteInput.focus();
+      if (this.state.minute) {
+        this.minuteInput.setSelectionRange(this.state.minute.length, this.state.minute.length);
+        event.preventDefault();
+      }
+    }
+  }
+
+  focusMeridiemFromSecond(event) {
+    // If the second is empty or the cursor is after the value, move focus to the meridiem.
+    if ((this.state.second.length === 0
+        || this.state.second.length === this.secondInput.selectionEnd)
+        && this.meridiemSelect
+    ) {
+      this.meridiemSelect.focus();
+      event.preventDefault();
+    }
+  }
+
   handleValueChange(event, type, timeValue, meridiem) {
     if (type === TimeUtil.inputType.HOUR) {
       this.setState({
         hour: timeValue,
         meridiem,
         hourInitialFocused: false,
       });
-    } else {
+    } else if (type === TimeUtil.inputType.MINUTE) {
       this.setState({
         minute: timeValue,
         minuteInitialFocused: false,
       });
+    } else {
+      this.setState({
+        second: timeValue,
+        secondInitialFocused: false,
+      });
     }
 
     // Input values of length 1 indicate incomplete time, which means we cannot get a
     // reliable time so onChange isn't triggered.
     if (this.props.onChange && timeValue.length !== 1) {
       const hour = type === TimeUtil.inputType.HOUR ? timeValue : this.state.hour;
       const minute = type === TimeUtil.inputType.MINUTE ? timeValue : this.state.minute;
+      const second = type === TimeUtil.inputType.SECOND ? timeValue : this.state.second;
 
-      if (hour === '' && minute === '') {
+      if (hour === '' && minute === '' && second === '') {
         this.props.onChange(event, '');
       } else {
-        this.props.onChange(event, this.formatHour(hour, meridiem).concat(':', minute));
+        this.props.onChange(event, this.formatHour(hour, meridiem).concat(':', minute).concat(this.props.showSeconds ? ':'.concat(second) : ''));
```

Comment:
If only the hour is present and both the minute and second are empty, this would concatenate two `:` to the hour and ends up with something like `12::`

Reply:
I'm not sure it should be changed. The util methods `splitHour`, `splitMinute`, `splitSecond` can be used to get the hour/minute/second if the colons are included. That is also the only place the passed in `onChange` callback is called so it could be considered a breaking change potentially to only call `onChange` if the time input is valid. Those are my thoughts.

Reply:
My comment was not about whether or not to call onChange. It should continue to call onChange just like it is now. My comment was about the formatted time value that is returned in the second parameter in onChange. If `showSeconds` is set, entering just the hour would return a formatted value consisting two colons with no minute and second (i.e. `12::`). Perhaps, this is fine as this indicates that a minute and second have not been entered.

Talked to @Blackop778 offline about this and decided to leave it as is for now unless anyone has other thoughts.